bitbots_utils/launch/record_bag.launch

```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<launch>
+    <arg name="sim" default="false" description="true: Use simulation time" />
+    <arg name="max_image_frequency" default="1.0" description="Max frequency [hz] for recording images" />
+    <arg name="max_pointcloud_frequency" default="1.0" description="Max frequency [hz] for recording pointclouds" />
+
+    <node name="record_rosbag_drop_images" pkg="topic_tools" exec="throttle"
+        args="messages /camera/image_proc $(var max_image_frequency) /camera/image_to_record" />
+    <node name="record_rosbag_drop_pointclouds" pkg="topic_tools" exec="throttle"
+        args="messages /line_mask_relative_pc $(var max_pointcloud_frequency) /line_mask_relative_pc_to_record" />
+
+    <!-- The command is a bit complicated, because we want to have a unique filename
+         and the date command would normally include a newline at the end. -->
+    <!-- We also set include-hidden-topics twice, because otherwise, the else case would produce some unprintable character that breaks the command. -->
+    <executable cmd="ros2 bag record \
+    -o $(env HOME)/rosbags/ID_$(env ROBOCUP_ROBOT_ID $(env ROBOT_NAME unknown_robot))_$(command 'bash -c &quot;date +%s |tr -d \\&quot;\\n\\&quot;&quot;') \
+    --node-name ros2_bag_record \
+    --include-hidden-topics \
+    --include-unpublished-topics \
+    --polling-interval 1000 \
+    $(eval '\'--use_sim_time\' if \'$(var sim)\'==\'true\' else \'--include-hidden-topics\'') \
```

Comment:
This is really horrible and unmaintainable. ROS 2 added Python launch files. They are specifically made to deal with more complicated logic. If you don't want to learn how to write them, I can do it. But I would really prefer if we don't merge this.

Reply:
ChatGPT converted it to this believable code. Will test it soon...
```py
from launch import LaunchDescription
from launch.actions import DeclareLaunchArgument, ExecuteProcess
from launch.substitutions import EnvironmentVariable, LaunchConfiguration, Command
from launch_ros.actions import Node

def generate_launch_description():
    return LaunchDescription([
        DeclareLaunchArgument(
            'sim', default_value='false', description='true: Use simulation time'),
        DeclareLaunchArgument(
            'max_image_frequency', default_value='1.0', description='Max frequency [hz] for recording images'),
        DeclareLaunchArgument(
            'max_pointcloud_frequency', default_value='1.0', description='Max frequency [hz] for recording pointclouds'),

        Node(
            package='topic_tools', executable='throttle', output='screen',
            name='record_rosbag_drop_images',
            arguments=[
                'messages', '/camera/image_proc',
                LaunchConfiguration('max_image_frequency'),
                '/camera/image_to_record'
            ]
        ),
        Node(
            package='topic_tools', executable='throttle', output='screen',
            name='record_rosbag_drop_pointclouds',
            arguments=[
                'messages', '/line_mask_relative_pc',
                LaunchConfiguration('max_pointcloud_frequency'),
                '/line_mask_relative_pc_to_record'
            ]
        ),

        ExecuteProcess(
            cmd=[
                'ros2', 'bag', 'record',
                '-o', EnvironmentVariable('HOME') + '/rosbags/ID_' +
                EnvironmentVariable('ROBOCUP_ROBOT_ID', default_value=EnvironmentVariable('ROBOT_NAME', default_value='unknown_robot')) +
                '_' + Command(['bash', '-c', '"date +%s |tr -d \\"\\n\\"']),
                '--node-name', 'ros2_bag_record',
                '--include-hidden-topics',
                '--include-unpublished-topics',
                '--polling-interval', '1000',
                '--use_sim_time' if LaunchConfiguration('sim')=='true' else '--include-hidden-topics',
                '/animation',
                '/audio/audio_info',
                '/audio/audio',
                '/ball_obstacle_active',
                '/ball_position_relative_filtered',
                '/ball_relative_filtered',
                '/ball_relative_movement',
                '/balls_relative',
                '/camera/camera_info',
                '/camera/image_to_record',
                '/clock',
                '/cmd_vel',
                '/cop_l',
                '/cop_r',
                '/core/power_switch_status',
                '/debug_markers',
                '/debug/approach_point',
                '/debug/ball_twist',
                '/debug/dsd/body_behavior',
                '/debug/dsd/hcm',
                '/debug/dsd/head_behavior',
                '/debug/dsd/localization',
                '/debug/used_ball',
                '/debug/which_ball_is_used',
                '/diagnostics',
                '/diagnostics_agg',
                '/DynamixelController/command',
                '/field_boundary_relative',
                '/game_controller_connected',
                '/gamestate',
                '/goal_pose',
                '/head_mode',
                '/imu_head/data',
                '/imu/data_raw',
                '/joint_states',
                '/line_mask_relative_pc_to_record',
                '/motion_odometry',
                '/move_base/current_goal',
                '/obstacles_relative',
                '/pose_with_covariance',
                '/robot_state',
                '/robots_relative',
                '/robots_relative_filtered',
                '/rosout',
                '/server_time_clock',
                '/speak',
                '/strategy',
                '/system_workload',
                '/team_data',
                '/tf',
                '/tf_static',
                '/time_to_ball'
            ],
            output='screen'
        ),
    ])
```

Reply:
Can we now change the (still quite hacky) `Command(['bash', '-c', '"date +%s |tr -d \\"\\n\\"'])` to a simple call to something like python's `time`? 

Also, I'm curious if we can now clean up`'--use_sim_time' if LaunchConfiguration('sim')=='true' else '--include-hidden-topics'` by completely omitting the parameter if the condition is not met. (Previously we had to resort  to an empty string which caused problems)